mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -372,4 +374,32 @@ public void getAnnotationsByKeysWithAllDepthShouldRetriveAnnotationsWithAllDepth
             .contains(PRIVATE_ANNOTATION, PRIVATE_CHILD_ANNOTATION, PRIVATE_GRANDCHILD_ANNOTATION);
     }
 
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionIfAnnotationDataIsOverLimitation() throws MailboxException {
+        expected.expect(AnnotationException.class);
+        Assume.assumeTrue(mailboxManager.hasCapability(MailboxCapabilities.Annotation));
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(MailboxAnnotation.newInstance(PRIVATE_KEY,
+            StringUtils.leftPad("AnyData", MailboxConstants.LIMIT_ANNOTATION_DATA + 1, '*'))));
+    }
+
+    @ContractTest
+    public void updateAnnotationsShouldThrowExceptionIfRequestCreateNewButMailboxIsOverLimit() throws MailboxException {
+        expected.expect(AnnotationException.class);
+        Assume.assumeTrue(mailboxManager.hasCapability(MailboxCapabilities.Annotation));
+        session = mailboxManager.createSystemSession(USER_2, LoggerFactory.getLogger("Test"));
+        MailboxPath inbox = MailboxPath.inbox(session);
+        mailboxManager.createMailbox(inbox, session);
+
+        ImmutableList.Builder<MailboxAnnotation> builder = ImmutableList.builder();
+        for (int i = 0; i < MailboxConstants.LIMIT_ANNOTATIONS_ON_MAILBOX; i++) {
+            builder.add(MailboxAnnotation.newInstance(new MailboxAnnotationKey("/private/comment" + i), "AnyValue"));
+        }
+        mailboxManager.updateAnnotations(inbox, session, builder.build());
+
+        mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(PRIVATE_ANNOTATION));
+    }
```

Comment:
You should verify that : 
 - You do not throw when you have the exact count of annotations
 - You have the exact size of annotations